src/command_modules/azure-cli-sql/azure/cli/command_modules/sql/custom.py

```diff
@@ -2071,3 +2075,161 @@ def managed_db_restore(
         managed_instance_name=target_managed_instance_name,
         resource_group_name=target_resource_group_name,
         parameters=kwargs)
+
+###############################################
+#              sql failover-group             #
+###############################################
+
+
+# pylint: disable=too-few-public-methods
+class FailoverPolicyType(Enum):
+    automatic = 'Automatic'
+    manual = 'Manual'
+
+
+def failover_group_create(
+        cmd,
+        client,
+        resource_group_name,
+        server_name,
+        failover_group_name,
+        partner_server,
+        partner_resource_group=None,
+        failover_policy=FailoverPolicyType.automatic.value,
+        grace_period=1,
+        add_db=None):
+
+    from six.moves.urllib.parse import quote  # pylint: disable=import-error
+    from azure.cli.core.commands.client_factory import get_subscription_id
+
+    # Build the partner server id
+    partner_server_id = "/subscriptions/{}/resourceGroups/{}/providers/Microsoft.Sql/servers/{}".format(
+        quote(get_subscription_id(cmd.cli_ctx)),
+        quote(partner_resource_group or resource_group_name),
+        quote(partner_server))
+
+    partner_server = PartnerInfo(id=partner_server_id)
+
+    # Convert grace period from hours to minutes
+    grace_period = int(grace_period) * 60
+
+    if add_db is None:
+        add_db = []
+
+    databases = _get_list_of_databases_for_fg(
+        get_subscription_id(cmd.cli_ctx),
+        resource_group_name,
+        server_name,
+        [],
+        add_db,
+        [])
+
+    return client.create_or_update(
+        resource_group_name=resource_group_name,
+        server_name=server_name,
+        failover_group_name=failover_group_name,
+        parameters=FailoverGroup(
+            partner_servers=[partner_server],
+            databases=databases,
+            read_write_endpoint=FailoverGroupReadWriteEndpoint(
+                failover_policy=failover_policy,
+                failover_with_data_loss_grace_period_minutes=grace_period),
+            read_only_endpoint=FailoverGroupReadOnlyEndpoint(
+                failover_policy="Disabled")))
+
+
+def failover_group_update(
+        cmd,
+        instance,
+        resource_group_name,
+        server_name,
+        failover_policy=None,
+        grace_period=None,
+        add_db=None,
+        remove_db=None):
+
+    from six.moves.urllib.parse import quote  # pylint: disable=import-error
+    from azure.cli.core.commands.client_factory import get_subscription_id
+
+    if failover_policy is not None:
```

Comment:
I don't have a strong preference, but elsewhere we abbreviate this to `if not failover_policy:` which the CLI reviewers had recommended.